From 4998d549be51e411f383e7bb6dfc7427f1ece804 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Tue, 21 Jul 2026 10:25:48 +0800
Subject: [PATCH] platform/loongarch: explicitly disables backlight before
 suspend

On some Loongson platforms, not disabling the backlight before suspend
can cause the system unable to resume the backlight when resuming.
---
 drivers/platform/loongarch/loongson-laptop.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/platform/loongarch/loongson-laptop.c b/drivers/platform/loongarch/loongson-laptop.c
index 61b18ac206c9e..1bdf2839b4104 100644
--- a/drivers/platform/loongarch/loongson-laptop.c
+++ b/drivers/platform/loongarch/loongson-laptop.c
@@ -189,6 +189,14 @@ static int __init setup_acpi_notify(struct generic_sub_driver *sub_driver)
 
 static int loongson_hotkey_suspend(struct device *dev)
 {
+	int ret;
+
+	ret = ec_backlight_set_power(target_powered);
+	if (ret < 0) {
+		pr_warn("loongson_backlight: failed to disable backlight (%d)\n", ret);
+		bl_powered = false;
+	}
+
 	return 0;
 }
 
-- 
2.52.0

